src/components/genesys-web-message/GenesysWebMessage.vue

```diff
@@ -8,22 +8,27 @@ import Vue from 'vue'
 import { Component, Prop } from 'vue-property-decorator'
 
 /**
- * This component creates the button to generate a chat with Service BC Helpdesk.
- * This component cannot be unit tested because it heavily relies on a third-party library.
+ * @property {string} genesysURL - the URL to the genesys library
+ * @property {string} environmentKey - the environment key (ID)
  * @property {string} deploymentKey - the deployment key (ID)
+ * Go To README.md for more details.
  */
 @Component({})
 export default class GenesysWebMessage extends Vue {
+  @Prop({ default: '' }) readonly genesysURL!: string
+
+  @Prop({ default: '0' }) readonly environmentKey!: string
+
   @Prop({ default: '0' }) readonly deploymentKey!: string
 
   mounted (): void {
     if (this.deploymentKey !== '0') {
```

Comment:
Should this check `genesysURL` and `environmentKey` as well?

Reply:
I actually wanted to put the default of these two their respective values (the url and environments provided). What do you think?

Reply:
So in the UI, if these two props weren't passed, the component would still work.

Reply:
No, don't make this component work "incorrectly" if the props aren't passed. These props are for a specific UI and environment.

Reply:
I'll add the check then, it's better.

Reply:
You could also simplify this a bit in one of two ways:

1. change `{ default: '0' }` to `{ default: '0' }`, and then you can simply check "if (environmentKey)"

2. change `{ default: '0' }` to `{ required: true }`, which forces the parent to provide the props, but then you'd still have to check that they're not "0"